numpy/core/_add_newdocs.py

```diff
@@ -5248,14 +5248,26 @@
 
     For a one-dimensional array, accumulate produces results equivalent to::
 
-      r = np.empty(len(A))
-      t = op.identity        # op = the ufunc being applied to A's  elements
-      for i in range(len(A)):
+      r = np.empty_like(A)
+      if use_identity():         # opaque identity decision
+          t = op.identity        # op = the ufunc being applied to A's elements
+          start = 0
+      else:
+          t = A[0]
+          r[0] = A[0]
+          start = 1
+          
+      for i in range(start, len(A)):
+          if use_identity():
+              t = op(t, op.identity)
```

Comment:
Are the lines 5261, 5262 correct? On every iteration `t = op(t, op.identity)` or only on the first iteration?

Reply:
Either and more: it could be never, on the first iteration, on all but the first iteration, always, etc. as determined by `use_identity`. The point is to illustrate that Numpy can choose to use the identity whenever it wants.

Reply:
But then this example sacrifices clarity for preciseness. I think it would be better to leave the pseudo-code as-is (with perhaps a mention that op.identity is 0 in NumPy.

Reply:
The pseudo-code, as is, is literally wrong. It's possible to show pseudocode that is simpler and right; if you want I can instead use this as an example. It will just be an implementation-dependent version of right that will require vigilance if this is to change.

Reply:
Here is the actual behaviour, which entirely disregards the identity:

```python
r = np.empty_like(A)
t = A[0]
r[0] = A[0]

for i in range(1, len(A)):
    t = op(t, A[i])
    r[i] = t
```

But see the discussion on the associated issue for nuance. If this simplified pseudocode is used, a disclaimer should be added that it is subject to change based on implementation and the identity might at some point be used.

Reply:
Having the pseudocode follow the implementation and adding a very short disclaimer is fine. If at some point we change the implementation, I would hope the reviewers would check the documentation.